scipy/stats/_rvs_sampling.py

```diff
@@ -133,7 +133,8 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
                "potentially causing bad performance.".format(int(exp_iter)))
         warnings.warn(msg, RuntimeWarning)
 
-    size1d = tuple(np.atleast_1d(size))
+    # convert to int, otherwise sth like size=1e3 does not work
+    size1d = tuple(np.atleast_1d(np.array(size, dtype=int)))
```

Comment:
While it is convenient, I don't think we should accept floating point values.  Numpy has moved away from allowing floating point values for arguments that are inherently integers (e.g. indexing, shapes, sizes), and I don't think any of the distributions accept floating point values in the `size` argument of the `rvs` method.

For exampe, `numpy.zeros`:

```
In [4]: np.zeros(1e2)                                                                                                 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-fa4503681fdc> in <module>
----> 1 np.zeros(1e2)

TypeError: 'float' object cannot be interpreted as an integer
```
The `size` argument of functions in `numpy.random`:
```
In [5]: np.random.uniform(size=1e2)                                                                                   
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-8e110be2a33b> in <module>
----> 1 np.random.uniform(size=1e2)

mtrand.pyx in mtrand.RandomState.uniform()

mtrand.pyx in mtrand.cont2_array_sc()

TypeError: 'float' object cannot be interpreted as an integer
```

The `size` argument of the `rvs` method of SciPy's distributions:
```
In [6]: from scipy.stats import gamma                                                                                 

In [7]: gamma.rvs(3.4, size=1e2)                                                                                      
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-877e4948d690> in <module>
----> 1 gamma.rvs(3.4, size=1e2)

~/miniconda36b/lib/python3.6/site-packages/scipy/stats/_distn_infrastructure.py in rvs(self, *args, **kwds)
    950         # by _rvs().
    951         self._size = size
--> 952         vals = self._rvs(*args)
    953 
    954         vals = vals * scale + loc

~/miniconda36b/lib/python3.6/site-packages/scipy/stats/_continuous_distns.py in _rvs(self, a)
   2514     """
   2515     def _rvs(self, a):
-> 2516         return self._random_state.standard_gamma(a, self._size)
   2517 
   2518     def _pdf(self, x, a):

mtrand.pyx in mtrand.RandomState.standard_gamma()

mtrand.pyx in mtrand.cont1_array_sc()

TypeError: 'numpy.float64' object cannot be interpreted as an integer
```

So this change should be dropped from the pull request.

Reply:
done, I had the impression that `rvs` does accept `1e3` etc, but since that is not the case, let's not introduce it here